Update Espressif border router code version (#149)
openthread/ot-docs: site/en/guides/border-router/espressif-esp32.md

Clone the [esp-idf](https://github.com/espressif/esp-idf) and the [esp-thread-br](https://github.com/espressif/esp-thread-br) repository.

```
$ git clone -b v5.1.2 --recursive https://github.com/espressif/esp-idf.git
$ cd esp-idf
$ ./install.sh
$ . ./export.sh
$ cd ..
$ git clone -b v1.0 --recursive https://github.com/espressif/esp-thread-br.git
```

> aside positive

> >

Reviewer: Please remove these two lines, you only need the one starting with `> Note:`

"aside positive" is only used in the codelabs format

Author: Done.

> > aside positive
> >
> > **Note:** `esp-thread-br` branch v1.0 with the `esp-idf` branch v5.1.2 is one of the release version, for the latest version of `esp-thread-br`, you can refer to [ESP Thread Border Router Codelab: 2.1.1. Set up the Repositories](https://docs.espressif.com/projects/esp-thread-br/en/latest/dev-guide/build_and_run.html#set-up-the-repositories).
> 

Reviewer: ```suggestion
> Note: The `esp-thread-br` branch v1.0 with the `esp-idf` branch v5.1.2 is one of the release versions. For the latest version of `esp-thread-br`, refer to [ESP Thread Border Router Codelab: 2.1.1. Set up the Repositories](https://docs.espressif.com/projects/esp-thread-br/en/latest/dev-guide/build_and_run.html#set-up-the-repositories).

```

Minor copyedit, plus the note format you had is for our codelabs markdown.  For markdown in other files, it needs to be this format: https://github.com/openthread/ot-docs/blob/main/STYLE_GUIDE.md#callouts
